hedera-node/hedera-config/src/main/java/com/hedera/node/config/data/FreezeUpgradeConfig.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.config.data;
+
+import com.swirlds.config.api.ConfigData;
+import com.swirlds.config.api.ConfigProperty;
+
+/**
+ * Defines the configuration needed for freeze upgrade.
+ * @param upgradeArtifactsPath directory path where upgrade artifacts are stored
+ */
+@ConfigData("freezeUpgrade")
+public record FreezeUpgradeConfig(
+        @ConfigProperty(defaultValue = "/opt/hgcapp/services-hedera/HapiApp2.0/data/upgrade/current")
+                String upgradeArtifactsPath) {}
```

Comment:
`Path` can be used as native type in config records.

Reply:
I tried to use Path in place of String for config record type, but it was not immediately obvious to me how to make this work. There are other config properties that use String paths. I think that I would like to leave it as String in this PR and file an issue to convert all String path config properties to Path type.